### PR TITLE
ENH: Styler css Str optional input arguments as well as List[Tuple]

### LINKED ISSUE #39563: ENH: Styler accepts css-strings input args as well as list of (attr,value) pairs
#### Is your feature request related to a problem?

Currently the `styler` has the ability to `set_table_styles` provided they are characterised with a css-selector and list of attribute value pairs, e.g.

```
styler.set_table_styles([{'selector': 'th', 'props': [('color', 'red'), ('font-size', '1em'), ('font-weight',  'bold')]}])
```

Even the legacy docs describe this as a bit of a pain, and for someone used to css, they will be typically looking for a css-style solution:

```
styler.set_table_styles([{'selector': 'th', 'props': 'color:red;font-size:1em;font-weight:bold;'}])
```

#### Describe the solution you'd like

optionally use the input argument valid CSS format above.

#### API breaking implications

none if backwards compatible.

### PR DESCRIPTION
**Pull request by attack68**
- [x] closes #39563 
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

**Comment by attack68** [comment]
this is ready for review..

File: pandas/io/formats/style.py [hunk 1]
```diff
@@ -2025,3 +2047,28 @@ def _maybe_wrap_formatter(
     else:
         msg = f"Expected a string, got {na_rep} instead"
         raise TypeError(msg)
+
+
+def _maybe_convert_css_to_tuples(
+    style: Union[str, Sequence[Tuple[str, Union[str, int, float]]]]
+):
```
Comment (jreback) [inline pandas/io/formats/style.py:2054]: can you add an output type (maybe define this at the top as CSSProperties or similar)

File: doc/source/whatsnew/v1.3.0.rst [hunk 2]
```diff
@@ -56,6 +56,7 @@ Other enhancements
 - :meth:`.Styler.set_tooltips` allows on hover tooltips to be added to styled HTML dataframes.
 - :meth:`Series.loc.__getitem__` and :meth:`Series.loc.__setitem__` with :class:`MultiIndex` now raising helpful error message when indexer has too many dimensions (:issue:`35349`)
 - :meth:`pandas.read_stata` and :class:`StataReader` support reading data from compressed files.
+- :meth:`.Styler.set_tooltips_class` and :meth:`.Styler.set_table_styles` amended to optionally allow certain css-string input arguments.
```
Comment (jreback) [inline doc/source/whatsnew/v1.3.0.rst:59]: locate after the other note, L56 (which is missing an issue number; use the PR number for it)

**Review by jreback — CHANGES_REQUESTED** [review]
looks fine. couple of comments.

**Comment by attack68** [comment]
@jreback changes made, with some better typing throughout the module.

one failed test is unrelated.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @attack68 keep em coming!

**Pull request merged**